docs/blocks/index.md

```diff
@@ -1,51 +1,182 @@
+        break;
+    }
+}
+finishingStage();
+```
+
+The following subsections further break down these stages into actual method calls.
+
+#### The "Initiating" Stage
+
+- `InputEvent.InteractionKeyMappingTriggered` is fired with the left mouse button and the main hand. If the event is canceled, the pipeline ends.
+- Several prerequisites are checked, for example that you are not in spectator mode, that all required feature flags for the `ItemStack` in your main hand are enabled or that the block in question is not outside the world border. If at least one of these checks fails, the pipeline ends.
+- `PlayerInteractEvent.LeftClickBlock` is fired. If the event is canceled, the pipeline ends.
+- `Block#attack` is called.
+
+#### The "Mining" Stage
+
+- `PlayerInteractEvent.LeftClickBlock` is fired. If the event is canceled, the pipeline moves to the "finishing" stage.
+- `Block#getDestroyProgress` is called and added to the internal destroy progress counter.
+    - `Block#getDestroyProgress` returns a float value between 0 and 1, representing how much the destroy progress counter should be increased every tick.
+- The progress overlay (cracking texture) is updated accordingly.
+- If the destroy progress is greater than 1.0 (i.e. completed, i.e. the block should be broken), the "mining" stage is exited and the "actually breaking" stage is entered.
+
+#### The "Actually Breaking" Stage
+
+- `Item#onBlockStartBreak` is called. If it returns `true`, the pipeline moves to the "finishing" stage.
+- Server-only: `IForgeBlock#canHarvestBlock` is called.
+- `Block#onDestroyedByPlayer` is called. If it returns `false`, the pipeline moves to the "finishing" stage. In that `Block#onDestroyedByPlayer` call:
+    - `Block#playerWillDestroy` is called.
+    - The blockstate is removed from the level via a `Level#setBlock` call with `Blocks.AIR.defaultBlockState()` as the blockstate parameter.
+        - In that `Level#setBlock` call, `Block#onRemove` is called.
+- `Block#destroy` is called.
+- Server-only: If the previous call to `IForgeBlock#canHarvestBlock` returned `true`, `Block#playerDestroy` is called.
+- Server-onlý: `IForgeBlock#getExpDrop` is called.
+- Server-only: `Block#popExperience` is called with the result of the previous `IForgeBlock#getExpDrop` call, if that call returned a value greater than 0.
+
+#### The "Finishing" Stage
+
+- The internal destroy progress counter is reset.
+
+### Ticking and Random Ticking
+
+Like many other things, blocks are ticked (updated) every tick, which is 1 / 20 of a second, or 50 milliseconds.
+
+There are two methods for ticking: `Block#tick` and `Block#animateTick`. The difference is that `Block#tick` is called only on the server, whereas `Block#animateTick` is called only on the client. For example, redstone updates are processed in `Block#tick`, while the fire particles of torches are spawned in `Block#animateTick`. Override either of them to add your server/client ticking behavior.
```

Comment:
Is `#tick` really called every tick? I don't think that's the case. I thought this is only called when someone schedules a tick update through the scheduler system.

Reply:
Yes, it is called every tick through ServerLevel#tick.

Reply:
So, I took a look, the method is *passed through* a tick block method which only ticks a block that has been scheduled on that specific tick. Think about it, if blocks tick every tick, what is the point of a block entity ticker method?

Reply:
I misinterpreted that specific code then.